pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexCRUDViolationRule.java

```diff
@@ -277,14 +296,69 @@ private void checkForCRUD(final AbstractApexNode<?> node, final Object data, fin
         if (variable != null) {
             final String type = varToTypeMapping.get(Helper.getFQVariableName(variable));
             if (type != null) {
-                StringBuilder typeCheck = new StringBuilder().append(node.getNode().getDefiningType().getApexName()).append(":")
-                        .append(type);
+                StringBuilder typeCheck = new StringBuilder().append(node.getNode().getDefiningType().getApexName())
+                        .append(":").append(type);
 
                 validateCRUDCheckPresent(node, data, crudMethod, typeCheck.toString());
             }
         }
     }
 
+    private HashSet<ASTMethodCallExpression> getPreviousMethodCalls(final AbstractApexNode<?> self) {
+        final HashSet<ASTMethodCallExpression> innerMethodCalls = new HashSet<>();
+
+        final ASTBlockStatement blockStatement = self.getFirstParentOfType(ASTBlockStatement.class);
+        if (blockStatement != null) {
+
+            List<ASTMethodCallExpression> nodes = blockStatement.findDescendantsOfType(ASTMethodCallExpression.class);
```

Comment:
This doesn't take order into account.

```
// Some DML

checkPerms();

// More DML
```

Would pass, but would still be wrong